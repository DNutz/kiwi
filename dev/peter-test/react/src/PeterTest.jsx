import React, { PureComponent, Fragment } from "react";
/**
 * This is PeterTest
 */
export default class PeterTest extends PureComponent {
    constructor(props) {
        super(props);

        /**
         * This is the state of the class
         * @type {state}
         * @private
         */
        this.state = {
            times: 1,
            color: "black"
        };
    }

    /**
     * I don't know what this does
     * @param {props} nextProps 
     * @param {state} prevState 
     */
    static getDerivedStateFromProps(nextProps, prevState) {
        if (nextProps.times !== prevState.times || nextProps.color !== prevState.color) {
            return {
                times: (nextProps.times !== undefined) ? ((Number(nextProps.times) >= 1) ? Number(nextProps.times) : prevState.times) : prevState.times,
                color: (nextProps.color !== undefined) ? nextProps.color : prevState.color
            };
        }

        return null;
    }

    /**
     * This is the render method
     * @returns The peter test repreated and in the color
     */
    render() {
        return (
            <Fragment>
                <h1 style={{fontFamily: "sans-serif"}}>Component: PeterTest</h1>
                {[...Array(this.state.times)].map((_, index) =>
                    <p key={index} style={{color: this.state.color, fontFamily: "sans-serif"}}>Look Peter the component works!</p>
                )}
            </Fragment>
        );
    }
};
