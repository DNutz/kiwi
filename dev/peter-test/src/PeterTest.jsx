import React, { PureComponent } from 'react';

export default class PeterTest extends PureComponent {
    constructor(props) {
        super(props);

        // The state variable holds our defaults and our application data (as it changes).
        this.state = {
            times: 1,
            color: 'black'
        };
    }

    // getDerivedStateFromProps is new in React 16.3. It will execute before the initial render and any subsequent times it needs to update.
    static getDerivedStateFromProps(nextProps, prevState) {
        // Only rerender if the color or times are different then before.
        if (nextProps.times !== prevState.times || nextProps.color !== prevState.color) {
            return {
                times: (nextProps.times !== undefined) ? ((Number(nextProps.times) >= 1) ? Number(nextProps.times) : prevState.times) : prevState.times,
                color: (nextProps.color !== undefined) ? nextProps.color : prevState.color
            };
        }

        return null;
    }

    render() {
        // [...Array(this.state.times)] is equivalent to saying 'repeat {this.state.times}'.
        // React needs a key for multiple identical children. The key must be unique as it distinguishes children nodes.
        return (
            [...Array(this.state.times)].map((_, index) =>
                <p key={index} style={{color: this.state.color}}>Look Peter the component works!</p>
            )
        );
    }
};