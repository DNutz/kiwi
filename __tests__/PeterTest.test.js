// PeterTest.react.test.js
import React from 'react';
//import PeterTest from '../PeterTest.react';
import PeterTest from '../dev/peter-test/src/PeterTest';
import renderer from 'react-test-renderer';

//test('PeterTest works', () => {
describe('PeterTest component renders the PeterTest correctly', () => {
  it('renders correctly', () => {
    const component = renderer.create(
        <PeterTest color="red" times="2"/>,
        //<h1>Hello World</h1>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});