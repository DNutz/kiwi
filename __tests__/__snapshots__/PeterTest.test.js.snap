// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeterTest component renders the PeterTest correctly renders correctly 1`] = `
Array [
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Look Peter the component works!
  </p>,
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Look Peter the component works!
  </p>,
]
`;
